library/Zend/Mvc/Controller/AbstractRestfulController.php

```diff
@@ -374,8 +375,12 @@ public function onDispatch(MvcEvent $e)
                     $id = null;
                 }
                 $action = 'head';
-                $this->head($id);
+                $headResult = $this->head($id);
                 $response = $e->getResponse();
+                if ($headResult instanceof HttpResponse) {
+                    $response->setStatusCode($headResult->getStatusCode());
```

Comment:
@Ocramius I thought about that, just didn't want to make that large of a change. I'm fine with adding the headers as well. I'm also fine with another solution all together.